app/grandchallenge/hanging_protocols/forms.py

```diff
@@ -116,3 +129,86 @@ def _validate_slice_plane_indicator(self, *, viewport, viewport_names):
                 error=f"Viewport {viewport['viewport_name']} has a slice_plane_indicator that is the same as the viewport_name.",
                 field="json",
             )
+
+
+class ViewContentExampleMixin:
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.instance:
+            interface_slugs = self.instance.interfaces.values_list(
+                "slug", flat=True
+            )
+
+            if interface_slugs.count() > 0:
+                self.fields[
+                    "view_content"
+                ].help_text += f"The following interfaces are used in your {self.instance._meta.verbose_name}: {oxford_comma(interface_slugs)}. "
+
+            view_content_example = self.generate_view_content_example()
+
+            if view_content_example:
+                self.fields[
+                    "view_content"
+                ].help_text += f"Example usage: {view_content_example}. "
+            else:
+                self.fields[
+                    "view_content"
+                ].help_text += "No interfaces of type (image, chart, pdf, mp4, thumbnail_jpg, thumbnail_png) are used. At least one interface of those types is needed to config the viewer. "
+
+        self.fields["view_content"].help_text += format_lazy(
+            'Refer to the <a href="{}">documentation</a> for more information',
+            reverse("documentation:detail", args=["viewer-content"]),
+        )
+
+    def generate_view_content_example(self):
+        images = list(
+            self.instance.interfaces.filter(
+                kind=InterfaceKindChoices.IMAGE
+            ).values_list("slug", flat=True)
+        )
+        mandatory_isolation_interfaces = list(
+            self.instance.interfaces.filter(
+                kind__in=InterfaceKind.interface_type_mandatory_isolation()
+            ).values_list("slug", flat=True)
+        )

```

Comment:
I think a simple list comprehension would be clearer here (it probably won't make a real difference if `interfaces` is a cached property, but it's easier to read):  

```suggestion
        images = [
           int.slug for int in self.instance.interfaces.filter(kind=InterfaceKindChoices.IMAGE)
        ]

        mandatory_isolation_interfaces = [
            int.slug for in in 
            self.instance.interfaces.filter(
                kind__in=InterfaceKind.interface_type_mandatory_isolation()
            )
        ]

```

Reply:
You're right here Anne. All of these should anyway be list comprehensions to save going to the database multiple times, this data is small enough to filter client side so is more efficient that way.

Reply:
If they become `cached_properties` like I suggested somewhere else, I think hitting the database is not a concern, since the filtering would be happening on the cached queryset, but regardless a list comprehension is cleaner.

Reply:
The queryset would be cached but not necessarily evaluated. Each time `filter` is called on it and the queryset evaluated (in this case by putting it in a list comprehension) you go back to the database. The suggestion here would result in two database hits, by doing a pure client side filtering you do it in one.

Reply:
Changed to list comprehension with client side filtering.
Done.